integration-tests/smoke_test.sh

```diff
@@ -3,38 +3,40 @@ set -e
 
 balanceInit=$(docker exec bor0 bash -c "bor attach /root/.bor/data/bor.ipc -exec 'Math.round(web3.fromWei(eth.getBalance(eth.accounts[0])))'")
 
-delay=600
-
-echo "Wait ${delay} seconds for state-sync..."
-sleep $delay
-
-
-balance=$(docker exec bor0 bash -c "bor attach /root/.bor/data/bor.ipc -exec 'Math.round(web3.fromWei(eth.getBalance(eth.accounts[0])))'")
-
-if ! [[ "$balance" =~ ^[0-9]+$ ]]; then
-    echo "Something is wrong! Can't find the balance of first account in bor network."
-    exit 1
-fi
-
-echo "Found matic balance on account[0]: " $balance
-
-if (( $balance <= $balanceInit )); then
-    echo "Balance in bor network has not increased. This indicates that something is wrong with state sync."
-    exit 1
-fi
-
-delayCheckpoint=300
-
-echo "Wait ${delayCheckpoint} seconds for checkpoint..."
-sleep $delayCheckpoint
-
-checkpointID=$(curl -sL http://localhost:1317/checkpoints/latest | jq .result.id)
-
-if [ $checkpointID == "null" ]; then
-    echo "Something is wrong! Could not find any checkpoint."
-    exit 1
-else
-    echo "Found checkpoint ID:" $checkpointID
-fi
-
+stateSyncFound="false"
+checkpointFound="false"
+
+while true
+do
+  
+    balance=$(docker exec bor0 bash -c "bor attach /root/.bor/data/bor.ipc -exec 'Math.round(web3.fromWei(eth.getBalance(eth.accounts[0])))'")
+
+    if ! [[ "$balance" =~ ^[0-9]+$ ]]; then
+        echo "Something is wrong! Can't find the balance of first account in bor network."
+        exit 1
+    fi
+
+    echo "Found matic balance on account[0]: " $balance
+
+    if (( $balance <= $balanceInit )); then
+        echo "Balance in bor network has not increased. Waiting for state sync..."
+    else
+        echo "State Sync occured!"
+        stateSyncFound="true"   
```

Comment:
Can we add a timer printing the seconds/minutes needed to get the stateSync?
I believe we can get good indicators by that

Reply:
🎉